ably/realtime_channel_spec_test.go

```diff
@@ -155,6 +157,1258 @@ func attachAndWait(t *testing.T, channel *ably.RealtimeChannel) {
+
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+
+		channel.Attach(ctx)
+
+		// Check that the attach message isn't sent
+
+		checkIfAttachSent := recorder.CheckIfSent(proto.ActionAttach, 1)
+		if attachSent := ablytest.Instantly.IsTrue(checkIfAttachSent); attachSent {
+			t.Fatalf("Attach message was sent before connection is established")
+		}
+
+		var channelStatechange ably.ChannelStateChange
+
+		ablytest.Instantly.Recv(t, &channelStatechange, channelStateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateAttaching, channelStatechange.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, channelStatechange)
+		}
+
+		ablytest.Instantly.NoRecv(t, nil, channelStateChanges, t.Fatalf) // Shouldn't receive ATTACH, message queued
+
+		close = c.To(
+			connecting,
+			connected,
+		)
+
+		// Check that the attach message is sent
+		if attachSent := ablytest.Instantly.IsTrue(checkIfAttachSent); !attachSent {
+			t.Fatalf("Should sent attach message, since connected")
+		}
+
+		defer safeclose(t, close)
+
+		ablytest.Soon.Recv(t, &channelStatechange, channelStateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateAttached, channelStatechange.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, channelStatechange)
+		}
+	})
+
+	t.Run("RTL4j1: AttachResume should be True when Attached", func(t *testing.T) {
+		t.Parallel()
+
+		app, err := ablytest.NewSandbox(nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		defer safeclose(t, app)
+
+		c, close := TransitionConn(t, nil, app.Options()...)
+		defer safeclose(t, close)
+
+		close = c.To(
+			connecting,
+			connected,
+		)
+
+		defer safeclose(t, close)
+
+		channelTransitioner := c.Channel("test")
+		channel := channelTransitioner.Channel
+
+		channelStateChanges := make(ably.ChannelStateChanges, 10)
+		channel.OnAll(channelStateChanges.Receive)
+
+		// channel state is initialized
+		if expected, got := ably.ChannelStateInitialized, channel.State(); expected != got {
+			t.Fatalf("expected %v; got %v", expected, got)
+		}
+
+		if channel.GetAttachResume() != false {
+			t.Fatalf("Channel attach resume should be false when channel state is INITIALIZED")
+		}
+
+		channelTransitioner.ToSpecifiedState(chAttaching, chAttached)
+
+		var channelStatechange ably.ChannelStateChange
+
+		ablytest.Instantly.Recv(t, &channelStatechange, channelStateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateAttaching, channelStatechange.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, channelStatechange)
+		}
+
+		ablytest.Instantly.Recv(t, &channelStatechange, channelStateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateAttached, channelStatechange.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, channelStatechange)
+		}
+		if channel.GetAttachResume() != true {
```

Comment:
RTL4j talks about the ATTACH_RESUME flag in the ATTACH message. We should check that, not the internal `attachResume` flag which isn't required (the spec just suggests it as an implementation).

Reply:
I think, I probably need to test another test for RTL4j

Reply:
Added missing test, resolved